runner.go

```diff
@@ -43,6 +44,7 @@ func newRuntime() *Runtime {
 	r.flags.StringVar(&configFile, "config", "", "Configuration file to use")
 	r.flags.BoolVar(&nonInteractive, "non-interactive", false, "Non-interactive mode")
 	r.flags.BoolVar(&listTasks, "list", false, "List all tasks")
+	r.flags.BoolVar(&interactive, "interactive", false, "Interactive mode")
```

Comment:
i don't understand the difference between noninteractive and interactive mode

they are two flags but they do very subtly different things, even though i would expect them to be inversions of each other

Reply:
The difference is taskrunner with no tasks specified defaults to watch mode on and taskrunner with tasks defaults to watchmode off. This makes the flags weird. If we are fine with changing the current default behavior we can just default taskrunner to tasks to watch mode as well.

Reply:
things i would expect:

`taskrunner` is default watch mode
`taskrunner --watch=false` runs taskrunner once (which is kind of bizarre)
`taskrunner footask` runs the thing once
`taskrunner --watch` runs the thing on watch

I don't expect anyone uses the non-interactive flag...?

Reply:
we can't change the default for a single task to watch though, since we run taskrunner tasks in CI

Reply:
I'd be fine then with what you said above but feel like we don't need support for 
"`taskrunner --watch=false` runs taskrunner once".

Reply:
Those expectations also make sense to me. @berfarah is there a reason to keep the non-interactive flag around?

Reply:
I'm happy for us to solve it that way. Let's make sure we document the behavior clearly

Reply:
i think this flag is annoying in the way it is because the go flags api doesn't support doing stuff like `--watch=false`--watch=true`. if we could get that to work we should do it.

Reply:
If I make it a string flag we could explicitly check for --watch=true/false. Is it worth doing that?

Reply:
sounds nice

Reply:
wait, does go not support boolean flags...?
https://gobyexample.com/command-line-flags

Reply:
@AChelikani can you try to get ^ to work. maybe there's something we overlooked before.